examples/thread_test/src/main.rs

```diff
@@ -0,0 +1,225 @@
+//! Inspired by the Rust standard TLS tests:
+//! <https://github.com/rust-lang/rust/tree/master/library/std/tests/thread_local>
+
+#![feature(thread_local)]
+
+use std::cell::{Cell, RefCell};
+use std::collections::HashMap;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::thread;
+use std::time::Duration;
+
+#[cfg(target_os = "hermit")]
+use hermit as _;
+
+// --- Main TLS demonstration ---
+
+// TLS variables of various types.
+thread_local! {
+	static TLS_VALUE: Cell<i32> = const { Cell::new(0) };
+	static TLS_U64: Cell<u64> = const { Cell::new(0) };
+	static TLS_F64: Cell<f64> = const { Cell::new(0.0) };
+	static TLS_BOOL: Cell<bool> = const { Cell::new(false) };
+	static TLS_CHAR: Cell<char> = const { Cell::new('A') };
+	static TLS_STRING: Cell<&'static str> = const { Cell::new("Initial String") };
+	static TLS_U8: Cell<u8> = const { Cell::new(0) };
+	static TLS_U64_2: Cell<u64> = const { Cell::new(0xdeadbeef) };
+}
+
+// A custom type with 16-byte alignment.
+#[derive(Clone, Copy)]
+#[repr(align(16))]
+struct AlignedType(u8);
+#[thread_local]
+static TLS_ALIGNED: Cell<AlignedType> = Cell::new(AlignedType(0x42));
+
+// Check that a pointer is 16-byte aligned.
+fn check_alignment<T>(ptr: *const T) {
+	let addr = ptr as *const u8 as usize;
+	assert!(
+		addr & 0xF == 0,
+		"Address {:#x} is not 16-byte aligned",
+		addr
+	);
+}
+
+// Global flag to detect TLS destructor execution.
+static TLS_DESTRUCTOR_RAN: AtomicBool = AtomicBool::new(false);
+
+// A type whose destructor sets the flag.
+struct DtorNotifier;
+impl Drop for DtorNotifier {
+	fn drop(&mut self) {
+		TLS_DESTRUCTOR_RAN.store(true, Ordering::SeqCst);
+	}
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
```

Comment:
This can just return `()`, right?

```suggestion
fn main() {
```